src/views/Graph.vue

```diff
@@ -689,26 +750,40 @@ export default {
      * @param {Object[]} edges
      * @param {{ [id: string]: SVGRect }} nodeDimensions
      */
-    async layout (nodes, edges, nodeDimensions) {
+    async layout (nodes, edges, nodeDimensions, cycles) {
       // generate the GraphViz dot code
-      const dotCode = this.getDotCode(nodeDimensions, nodes, edges)
+      const dotCode = this.getDotCode(nodeDimensions, nodes, edges, cycles)
 
       // run the layout algorithm
       const jsonString = (await this.graphviz).layout(dotCode, 'json')
       const json = JSON.parse(jsonString)
 
+      this.subgraphs = {}
```

Comment:
I used `map` to update the graph edges because I guessed that rebuilding the set in a single operation might play more nicely with Vue's reactivity than emptying and rebuilding in a series of steps.

This guess might be completely wrong, anyone know different?

Reply:
I wouldn't have thought that would matter, as surely this code has to finish executing synchronously before any Vue reactivity can come into play.

Generally I do favour using `Array.prototype.map` for its elegance (and it's surely more efficient than creating an empty array and then pushing to it which means the JS engine will keep having to increase the size of the array). But in this case as we can iterate over `json.objects` once to extract the subgraphs and the node transformations I think it makes sense to do it the current way.

Reply:
Fine by me.

> this code has to finish executing synchronously before any Vue reactivity can come into play.

Vue reacts to changes asynchronously, however I presume that Vue's detection of changes must somehow boil down to synchronous callbacks registered on reactive things. E.G. replacing an array in-situ might result in one "array modified" callback, but adding N items to the array might result in N "item added" callbacks. At a later time Vue then responds to the changes captured by these callbacks.